setup/bundles.rst

```diff
@@ -198,6 +198,56 @@ adding a ``setDefined()`` method. The recommended check in this case would be::
     to detect the version of the ``symfony/http-kernel`` component, because it
     is the component where this constant is defined.
 
+Generic conditions for component version detection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You should strive to detect the exact feature that you want to use as
+soon as it is available. That is not always possible however, so here is
+a list of generic conditions you may use for lack of a better
+alternative. These conditions detect new APIs as soon as they are
+available.
+
+2.6
+...
+
+OptionsResolver
+"""""""""""""""
+
+.. code-block:: php
+
+    method_exists(Symfony\Component\OptionsResolver\OptionsResolver, 'setDefined')
```

Comment:
Using FQCNs here makes it compatible with old versions of php, but even if you are guaranteed to have access to `::class`, this notation might be better because detection tricks are meant to be removed in the next major version of the bundle they appear in, and this will make the removal less noisy.